File: app/src/main/java/com/ataulm/whatsnext/api/ApiSearchResponse.java
package com.ataulm.whatsnext.api;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class ApiSearchResponse {

    @SerializedName("next")
    public String cursorToNextPageOfResults;

    @SerializedName("items")
    public List<Result> searchItems;

    public static class Result {

        @SerializedName("type")
        public String type;

        @SerializedName("film")
        public ApiFilmSummary filmSummary;
    }
}

File: letterboxd/src/main/java/com/ataulm/letterboxd/ApiSearchResponse.kt
package com.ataulm.letterboxd

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class ApiSearchResponse(
    @field:Json(name = "next")
    val cursorToNextPageOfResults: String,

    @field:Json(name = "items")
    val searchItems: List<Result>
) {

    @JsonClass(generateAdapter = true)
    class Result(
        @field:Json(name = "type")
        val type: String,

        @field:Json(name = "film")
        val filmSummary: ApiFilmSummary
    )
}
